pkg/object/autocertmanager/autocertmanager.go

```diff
@@ -55,13 +53,34 @@ var aliases = []string{
 func init() {
 	supervisor.Register(&AutoCertManager{})
 	api.RegisterObject(&api.APIResource{
-		Category: Category,
-		Kind:     Kind,
-		Name:     strings.ToLower(Kind),
-		Aliases:  aliases,
+		Category:    Category,
+		Kind:        Kind,
+		Name:        strings.ToLower(Kind),
+		Aliases:     aliases,
+		ValiateHook: validateHook,
 	})
 }
 
+func validateHook(operationType api.OperationType, spec *supervisor.Spec) error {
+	if operationType != api.OperationTypeCreate || spec.Kind() != Kind {
+		return nil
+	}
+
+	acms := []string{}
+	supervisor.GetGlobalSuper().WalkControllers(func(controller *supervisor.ObjectEntity) bool {
+		if controller.Spec().Kind() == Kind {
+			acms = append(acms, controller.Spec().Name())
+		}
+		return true
+	})
+
+	if len(acms) >= 1 {
+		return fmt.Errorf("only one AutoCertManager is allowed, existed: %v", acms)
+	}
+
+	return nil
+}
+
```

Comment:
Do we need to add a lock for this hook? I mean in a really specially case, if two users create `AutoCertManger` at same time, so they may call this function at same time, then they may both create it successfully. 

Of course, in practice, it may never happen, usually only one administrator creates or deletes Easegress objects. Just some defensive programming advice.

Reply:
Your concern has been fixed in API layer, we call validate hooks within a global lock. You can see the change in api/object.go

Reply:
see that. you are right